src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,733 @@
+     * @param operator the operator to get the total magnitude for
+     * @param strategies the strategies to get the total magnitudes for
+     * @return totalMagnitudes the total magnitudes for each strategy
+     */
+    function getTotalMagnitudes(
+        address operator,
+        IStrategy[] calldata strategies
+    ) external view returns (uint64[] memory) {
+        uint64[] memory totalMagnitudes = new uint64[](strategies.length);
+        for (uint256 i = 0; i < strategies.length; ++i) {
+            (bool exists,, uint224 value) = _totalMagnitudeUpdate[operator][strategies[i]].latestSnapshot();
+            if (!exists) {
+                totalMagnitudes[i] = WAD;
+            } else {
+                totalMagnitudes[i] = uint64(value);
+            }
+        }
+        return totalMagnitudes;
+    }
+
+    /**
+     * @notice Returns the total magnitudes of an operator for a given set of strategies at a given timestamp
+     * @param operator the operator to get the total magnitude for
+     * @param strategies the strategies to get the total magnitudes for
+     * @param timestamp the timestamp to get the total magnitudes at
+     * @return totalMagnitudes the total magnitudes for each strategy
+     */
+    function getTotalMagnitudesAtTimestamp(
+        address operator,
+        IStrategy[] calldata strategies,
+        uint32 timestamp
+    ) external view returns (uint64[] memory) {
+        uint64[] memory totalMagnitudes = new uint64[](strategies.length);
+        for (uint256 i = 0; i < strategies.length; ++i) {
+            (uint224 value, uint256 pos) = _totalMagnitudeUpdate[operator][strategies[i]].upperLookupWithPos(timestamp);
+            // if there is no existing total magnitude snapshot
+            if (value == 0 && pos == 0) {
+                totalMagnitudes[i] = WAD;
+            } else {
+                totalMagnitudes[i] = uint64(value);
+            }
+        }
+        return totalMagnitudes;
+    }
+
+    /**
+     * @notice Returns the current total magnitude of an operator for a given strategy
+     * @param operator the operator to get the total magnitude for
+     * @param strategy the strategy to get the total magnitude for
+     * @return totalMagnitude the total magnitude for the strategy
+     */
+    function getTotalMagnitude(address operator, IStrategy strategy) external view returns (uint64) {
+        uint64 totalMagnitude;
+        (bool exists,, uint224 value) = _totalMagnitudeUpdate[operator][strategy].latestSnapshot();
+        if (!exists) {
+            totalMagnitude = WAD;
+        } else {
+            totalMagnitude = uint64(value);
+        }
+
+        return totalMagnitude;
+    }
+
+    /**
+     * @notice Returns the latest pending allocation of an operator for a given strategy and operatorSets.
+     * One of the assumptions here is we don't allow more than one pending allocation for an operatorSet at a time.
```

Comment:
should we consider asserting in this function that the constant used elsewhere has value 1 or something? or perhaps just add this to testing

Reply:
looks like the same is true of `getPendingDeallocations`

Reply:
Refactored to use `_getPendingMagnitudeInfo` throughout